File: Object Pool Pattern/PHP/Pool.php
<?php

require_once "Obj.php";

class Pool 
{
    protected static $objects = [];

    public static function initialize(int $number = 5)
    {
        for ($i = 0; $i < $number; $i++) {
            $id = uniqid("OBJ");
            self::$objects[$id] = ['item' => new Obj($id), 'status' => 1];
        }
    }

    public static function checkIn($data)
    {
        foreach(array_keys(self::$objects) as $key) {
            if(self::$objects[$key]['status'] == 1) {
                self::$objects[$key]['status'] = 0;
                return self::$objects[$key]['item']->resetWith($data);
            }
        }
        return NULL;
    }

    public static function checkOut(Obj $obj)
    {
        self::$objects[$obj->getID()]['status'] = 1;
    }
}

?>

File: Object Pool Pattern/CS/Pool.cs
using System.Collections.Generic;

namespace ObjectPool
{
    public sealed class Pool
    {
        private Dictionary<int, IObject> _objects;

        public Pool(int number = 5)
        {
            _objects = new Dictionary<int, IObject>(number);

            for (var i = 0; i < number; i++)
            {
                var id = GetHashCode();
                _objects.Add(id, new IObject(id, null));
            }
        }

        public IObject CheckIn(string data)
        {
            foreach (var key in this._objects.Keys)
            {
                var temp = this._objects[key].SetData(data);
                this._objects.Remove(key);
                return temp;
            }

            return null;
        }

        public IObject CheckOut(IObject obj)
        {
            this._objects.Add(obj.GetCode(), obj);
            return null;
        }
    }
}